### PR TITLE
Documentation about metrics definition

### PR DESCRIPTION
**Pull request by bcolloran**
@godzig and @magorlick could you guys take a look at these docs? I'm sure there's a typo or two buried in here, but I also haven't worked with our documentation tools so I don't know how the formatting and stuff like that works—let me know if I'm on the wrong track here or if there are style guides or something that should be followed for documentation like this.

Marissa, also looking to you for any thoughts about the actual content regarding measures in specific.

### CODE REVIEW

**Review by magorlick — COMMENTED** [review]
This is such a great document - thank you for writing it!  I left lots of small changes across the doc trying to position this more around the dashboard and linking out to duckDB resources.  I also made some changes to the H1s to make them H2-H3 for docusaurus.  It doesn't show H1s except the first.

File: docs/docs/defining-metrics.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+# Measures
+measures are numeric aggregates of columns from your data model, and power the line charts that you see in Rill.
+
+A measure must be defined with a [DuckSQL](./sqldialects/duck-sql.md) aggregation function over columns from your data model, or a mathematical expression built with one or more such aggregates.
+
+For example, if you have a table of sales events with columns including a timestamp for the sales date, the sales price, and customer id, you could calculate the following metrics per time period with these expressions:
+* number of sales: `COUNT(*)` (note that this would be equivalent to counting any column, e.g. `COUNT(sales_date)`)
+* total revenue: `SUM(sales_price)` 
+* revenue per customer: `SUM(sales_price)/COUNT(customer_id)`
```
[the remark this answers is not in the document]
Comment (bcolloran) [reply docs/docs/defining-metrics.md:39]: Whoops thanks!

File: docs/docs/defining-metrics.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+---
+title: Defining Metrics
+---
+
+In Rill, Your dashboards are powered by _metrics_. Metrics are composed of:
+* a _time dimension_, which will be a time stamp column from your data model. The time dimension will be the x-axis in the line charts shown on your dashboard, as well as serving as the grouping dimension over which _measure_ aggregates are computed.
+
+* _measures_ which are numerical aggregates of columns from your data model, and are shown on the y-axis of the line charts in your dashboard as well as the "big number" summaries next to the line charts. 
+
+* _dimensions_ which are categorical columns from your data model. Dimensions are shown in _leaderboards_, and allow you to filter the data shown in your dashboard.
```
[the remark this answers is not in the document]
Comment (bcolloran) [reply docs/docs/defining-metrics.md:10]: This one is a little bit unclear to me, I'm going to pass on incorporating this into the first batch of changes, let's discuss it more.

File: docs/docs/defining-metrics.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+Your time dimension must be a column from your data model of type [`TIMESTAMP`](https://duckdb.org/docs/sql/data_types/timestamp), [`TIME`](https://duckdb.org/docs/sql/data_types/overview), or [`DATE`](https://duckdb.org/docs/sql/data_types/date).
+
+:::tip
+
+Strings representing dates are not supported, but you may be able to [`CAST`](https://duckdb.org/docs/sql/expressions/cast) such a string to one of these types while developing your data model.
+
+:::
+
+### Measures
+measures are numeric aggregates of columns from your data model, and power the line charts that you see in Rill.
```
Comment (magorlick) [inline docs/docs/defining-metrics.md:36]:
```suggestion
Measures are numeric aggregates of columns from your data model, and power the line charts that you see in Rill.
```

**Review by magorlick — APPROVED** [review]

**Pull request merged**